### PR TITLE
Add some dependencies, and divide the client definition into another file.

### PR DESCRIPTION
**Pull request by Krout0n**
Related #11  #19. After this PR is merged, I'm gonna work on #11 .

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+use async_trait::async_trait;
+
+// This trait is defined because of mocking tests.
+#[async_trait]
+pub trait Clientable {
+    // see: https://github.com/Krout0n/mackerel-client-rs/blob/e49fc4fb535363db30aa876ccbe030ad7b59a8b0/src/host.rs#L129-L138
+    async fn post_metrics(
```
Comment (Krout0n) [inline src/client.rs:7]: See: https://crates.io/crates/async-trait
Trait-fns can't contain asyncness. So, async-trait is required. (rustc suggests that.)

File: src/client.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+use async_trait::async_trait;
+
+// This trait is defined because of mocking tests.
+#[async_trait]
```
Comment (ne-sachirou) [inline src/client.rs:4]: note: This https://github.com/dtolnay/async-trait .

**Review by ne-sachirou — APPROVED** [review]
🙆‍♀️

**Comment by Krout0n** [comment]
Thank you!

**Pull request merged**